tests/test/scheduler/test_scheduler.cpp

```diff
@@ -783,4 +804,30 @@ TEST_CASE_METHOD(SlowExecutorFixture,
       std::get<3>(actualTuple);
     REQUIRE(actualDiffs.size() == diffs.size());
 }
+
+TEST_CASE_METHOD(DummyExecutorFixture, "Test executor reuse", "[scheduler]")
```

Comment:
This is the sense check I mention in the comment. It was the first thing I wanted to double check when looking into this issue.